bpf/bpf_host.c

```diff
@@ -522,8 +524,10 @@ handle_ipv4(struct __ctx_buff *ctx, __u32 secctx,
 		return ipv4_local_delivery(ctx, ETH_HLEN, secctx, ip4, ep,
 					   METRIC_INGRESS, from_host);
 	}
-
 #ifdef ENCAP_IFINDEX
+	else if (!from_host)
+		return CTX_ACT_OK;
```

Comment:
This early exit will lead us to skip the `redirect()` below (line 558). That redirect was introduced by @aanm in https://github.com/cilium/cilium/commit/f54f15d289143b01d020336a2ecb0c4572df0e55 and I'm unable to say if we can actually skip it without issues. If we can, should we also skip it in the native routing case as for IPv6?

Note that `redirect()` doesn't exist in the IPv6 path.

Reply:
We should probably remove that given this only relates to flannel support. Do we still actually support or test that @aanm? If its of concern, I can also disable bpf host routing in case of flannel.

Reply:
I added a patch to disable bpf host routing for flannel which should address this here.

Reply:
I think it might be not enough just disable the BPF routing when the flannel mode is detected, as https://github.com/cilium/cilium/commit/f54f15d289143b01d020336a2ecb0c4572df0e55 might be required by the CNI chaining (!= the flannel mode). @aanm Could you confirm that?

Reply:
Ah, from the quick grepping it seems that the commit message is a bit misleading, and `HOST_REDIRECT_TO_INGRESS` is set only when running in the flannel mode. So, ignore my comment above.